### PR TITLE
Fix container app create not waiting for ready environment

### PR DESCRIPTION
**Pull request by p-bouchon**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by zhoxing-ms** [comment]
@p-bouchon Please take a look at this comment https://github.com/Azure/azure-cli-extensions/pull/6399#issuecomment-1589564154 first.

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_workload_profile_commands.py [hunk 1]
```diff
@@ -244,6 +244,69 @@ def test_containerapp_create_with_workloadprofile_yaml(self, resource_group):
         clean_up_test_file(containerapp_file_name_01)
         clean_up_test_file(containerapp_file_name_02)
 
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer(location="eastus")
+    @live_only()  # encounters 'CannotOverwriteExistingCassetteException' only when run from recording (passes when run live)
```
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_workload_profile_commands.py:249]: May I ask why we need to mark this test as `@live_only()`? I guess this problem may require you to upload the recorded yaml file to solve it

File: src/containerapp/azext_containerapp/custom.py [hunk 2]
```diff
@@ -473,6 +473,11 @@ def create_containerapp(cmd,
     if not managed_env_info:
         raise ValidationError("The environment '{}' does not exist. Specify a valid environment".format(managed_env))
 
+    while not no_wait and "properties" in managed_env_info and managed_env_info["properties"]["provisioningState"].lower() in ["inprogress", "updating"]:
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/custom.py:476]: Recomment to use safe_get

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_workload_profile_commands.py (hunk 1 above)
Comment (p-bouchon) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_workload_profile_commands.py]: incorrectly marked, fixed

File: src/containerapp/azext_containerapp/custom.py [hunk 3]
```diff
@@ -473,6 +473,11 @@ def create_containerapp(cmd,
...
+        logger.info("Waiting for environment provisioning to finish before creating container app")
+        time.sleep(10)
```
[the remark this answers is not in the document]
Comment (p-bouchon) [reply src/containerapp/azext_containerapp/custom.py]: lowered a bit

File: src/containerapp/azext_containerapp/custom.py [hunk 4]
```diff
@@ -473,9 +473,9 @@ def create_containerapp(cmd,
     if not managed_env_info:
         raise ValidationError("The environment '{}' does not exist. Specify a valid environment".format(managed_env))
 
-    while not no_wait and "properties" in managed_env_info and managed_env_info["properties"]["provisioningState"].lower() in ["inprogress", "updating"]:
+    while not no_wait and safe_get(managed_env_info, "properties", "provisioningState").lower() in ["inprogress", "updating"]:
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/custom.py:476]: if `safe_get(managed_env_info, "properties", "provisioningState")` return `None`, the `lower()` will throw error.

Recommend to set a default empty string:
`safe_get(managed_env_info, "properties", "provisioningState", default="")`

**Review by Greedygre — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**